pkgs/development/compilers/graalvm/community-edition/mkGraal.nix

```diff
@@ -85,7 +85,7 @@ let
   ] ++ lib.optionals useMusl [
     (lib.getDev musl)
     # GraalVM 21.3.0+ expects musl-gcc as <system>-musl-gcc
-    (writeShellScriptBin "${stdenv.system}-musl-gcc" ''${lib.getDev musl}/bin/musl-gcc "$@"'')
+    (writeShellScriptBin "${stdenv.hostPlatform.system}-musl-gcc" ''${lib.getDev musl}/bin/musl-gcc "$@"'')
```

Comment:
what's the difference between those 2?

Reply:
`stdenv.hostPlatform` is the platform where the code is going to run. This is different from the `stdenv.buildPlatform` that is the code where the code is building.

It doesn't make any difference here, because GraalVM doesn't support cross compilation. But `stdenv.system` AFAIK, is deprecated (because it is not clear what is happening).